app: Clipboard events API fallback for pastes

Diamondtroller opened the pull request:
Async Clipboard API isn't fully supported on Firefox browser ([see MDN Web Docs](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard_API#browser_compatibility)). This patch implements **reading** clipboard data through Clipboard Event API as fallback in that scenario.

Diamondtroller committed:
app,cleanup: Support for firefox paste and other tweaks

Since Firefox's async Clipboard API isn't enabled for pastes, make non-async one be used. In other tweaks linting rules defined and hostname extraced from request literals.

Diamondtroller committed:
api: add protocol property and make it https

quackbarc commented on src/sketch-app/sketch-app.html:
```diff
@@ -1477,21 +1482,7 @@ <h2>Etch-a-Sketch</h2>
-            };
-
-            this.$.sketch.stop();
-            this.$.sketch.data = text;
-            this.peekID = null;
-            this.sketchDisabled = false;
-            this._alert('pasted from clipboard');
-          })
+          .then(this._pasteProto.bind(this))
```
Hmm, I'm not sure if it should _just_ be this hunk that's moved to a new method, cuz some other stuff would be left out like the return cases...

quackbarc replied:
Oh! You could also turn that `.then` statement into
```js
.then(text => this._pasteProto(text))
```
just to be consistent with that `.catch` right below it

quackbarc commented on src/sketch-app/sketch-app.html:
```diff
@@ -1501,6 +1492,22 @@ <h2>Etch-a-Sketch</h2>
           });
       },
 
+      _pasteProto: function(text) {
```
This _could_ be renamed to `_pasteText` or something, I think!

quackbarc reviewed (commented):
This looks fine! I'm not sure if I'm into the idea of making a new `_paste` method just yet though, but if it works it works, hehe

I think you also left out your changes to polymer.json and sketch-api.html on the PR though... but I wouldn't mind adding that polymer.json change! Anyways...

Diamondtroller committed:
app,api: Handle clipboard read error nicer and implement req. changes

Since _paste() and _pasteText() both fire on paste while using
keyboard shortcuts, implement a check to not show the error.
Additionally remove address interpolation of HTTP requests as
being out of scope.

quackbarc commented:
GRAAAAH I took way too long to review this damn PR!! Lettuce Glitter Tomato Mayonnaise or whatever people say in those PRs !

---

(That revert btw was an accident. I completely forgot to push other things I had onto master first before pushing the PR. Oh well)
